File: offer/JZ68 二叉搜索树的最近公共祖先/tree.java
import java.util.*;

/*
 * public class TreeNode {
 *   int val = 0;
 *   TreeNode left = null;
 *   TreeNode right = null;
 *   public TreeNode(int val) {
 *     this.val = val;
 *   }
 * }
 */

public class Solution {
    private TreeNode find(TreeNode root, int a, int b){
        if(root == null)
            return null;
        if(a <= root.val && root.val <= b)
            return root;
        if(a < root.val && b < root.val)
            return find(root.left, a, b);
        return find(root.right, a, b);
    }
    public int lowestCommonAncestor (TreeNode root, int a, int b) {
        if(a > b){
            int tmp = a;
            a = b;
            b = tmp;
        }
        return find(root, a, b).val;
    }
}

File: 0102. Binary Tree Level Order Traversal/tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr){
            return {};
        }
        queue<TreeNode*> q;
        q.push(root);
        vector<vector<int>> ret;
        while(!q.empty()){
            vector<int> cur;
            int size = q.size();
            while(size > 0){
                TreeNode* node = q.front();
                q.pop();
                cur.push_back(node->val);
                if(node->left != nullptr)
                    q.push(node->left);
                if(node->right != nullptr)
                    q.push(node->right);
                size--;
            }
            ret.push_back(cur);
        }
        return ret;
    }
};